internal/cortex/querier/queryrange/results_cache.go

```diff
@@ -357,6 +360,45 @@ func (s resultsCache) isAtModifierCachable(r Request, maxCacheTime int64) bool {
 	return atModCachable
 }
 
+var negativeOffset = errors.New("negative offset")
+
+func (s resultsCache) isOffsetCachable(r Request) bool {
```

Comment:
Better add a comment about we don't cache negative offset but still cache postive offset